File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    released = [False]*len(progresses)
    left = cal_completed_days(progresses, speeds)

    i = 0
    while i < (len(progresses)):
        cnt = 0
        if released[i] == False:
    
            day = left[i]
            released[i] = True
            cnt += 1
            
            while i<len(progresses)-1:
                if left[i+1] > day:
                    break
                else:
                    released[i+1] = True
                    cnt += 1
                    i += 1
        i+=1
        print("cnt: ",cnt)
        answer.append(cnt)

    return answer

def cal_completed_days(progresses,speeds):
    left = []
    
    for i in range(len(progresses)):
        day = 100-progresses[i]
        left.append(math.ceil(day/speeds[i]))
    return left

File: 프로그래머스/2/42586. 기능개발/기능개발.swift
import Foundation

func solution(_ progresses: [Int], _ speeds: [Int]) -> [Int] {
    var numOfReleases: [Int] = []
    var left: [Int] = []
    
    for i in 0..<progresses.count {
        left.append(Int(ceil(Double(100-progresses[i])/Double(speeds[i]))))
    }
    
    while !left.isEmpty {
        var cnt = 1
        var lastReleaseDate = left.removeFirst()
        while true {
            if lastReleaseDate >= left.first ?? 101 {
                left.removeFirst()
                cnt += 1
            } else {
                break
            }
        }
        numOfReleases.append(cnt)
    }
    return numOfReleases
}